### PR TITLE
Add nvcc 11.5 to CI

### LINKED ISSUE #1475: Add CI Jobs for CUDA 11.5
We should add CI Jobs for CUDA 11.5.

_Originally posted by @j-stephan in https://github.com/alpaka-group/alpaka/pull/1466#discussion_r753104453_

### PR DESCRIPTION
**Pull request by bernhardmgruber**
Fixes #1475

### CODE REVIEW

**Comment by psychocoderHPC** [comment]
Why do we not add it to the HZDR CI. We are working on removing such jobs from the workflows, IMO this is doubling the work.

**Comment by bernhardmgruber** [comment]
> Why do we not add it to the HZDR CI. We are working on removing such jobs from the workflows, IMO this is doubling the work.

Because I have less experience with the GitLab CI and nobody else did it. But please, go ahead and add the GitLab CI jobs! I can close this PR when yours is ready.

**Comment by psychocoderHPC** [comment]
It is equal to adding it to git workflows. The difference is that it is not on one line.

https://github.com/alpaka-group/alpaka/blob/ac1ccf380d4208fbff8e5c592f649c569a9a308a/script/gitlabci/job_cuda.yml#L2-L12

Best is to have a base job file per cuda version and than add it to .gitlab.yml but you removed it in one of the last PRs.

**Comment by psychocoderHPC** [comment]
@SimeonEhrig mentioned yesterday that NVIDIA is currently not providing CUDA 11.5 docker containers because they have issues building them. The HZDR CI is using CUDA docker container as a base to avoid driver issue I got with normal ubuntu container when installing CUDA via apt only.
So we will merge this one before #1491 and I will work on fixing the dependency to NVIDIA docker container in the HZDR CI if NVIDIA is not providing docker container before I can work on this topic.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -406,6 +406,43 @@ jobs:
           os: ubuntu-latest
           env: {CXX: clang++, CC: clang,  ALPAKA_CI_CLANG_VER: 12,     ALPAKA_CI_STDLIB: libstdc++, CMAKE_BUILD_TYPE: Release, ALPAKA_CI_BOOST_BRANCH: boost-1.72.0, ALPAKA_CI_CMAKE_VER: 3.19.7,                     ALPAKA_CI_DOCKER_BASE_IMAGE_NAME: "ubuntu:20.04", ALPAKA_ACC_GPU_CUDA_ENABLE: ON, ALPAKA_CI_CUDA_VERSION: "11.4", CMAKE_CUDA_COMPILER: nvcc, CMAKE_CUDA_ARCHITECTURES: "86",              ALPAKA_ACC_CPU_B_SEQ_T_FIBERS_ENABLE: OFF}
 
+        ## CUDA 11.5
```
Comment (psychocoderHPC) [inline .github/workflows/ci.yml:388]: Could you please update the build matrix in README.md .

**Review by psychocoderHPC — CHANGES_REQUESTED** [review]

**Comment by SimeonEhrig** [comment]
> @SimeonEhrig mentioned yesterday that NVIDIA is currently not providing CUDA 11.5 docker containers because they have issues building them. The HZDR CI is using CUDA docker container as a base to avoid driver issue I got with normal ubuntu container when installing CUDA via apt only. So we will merge this one before #1491 and I will work on fixing the dependency to NVIDIA docker container in the HZDR CI if NVIDIA is not providing docker container before I can work on this topic.

https://gitlab.com/nvidia/container-images/cuda/-/issues/142

**Review by psychocoderHPC — DISMISSED** [review]

**Review by psychocoderHPC — APPROVED** [review]

**Pull request merged**